### DISCUSSION TITLE
Can copilot lab support http proxy?

### DISCUSSION THREAD

**Discussion by Novelfor** [community/community#49294] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I'm using VSCode in remote-ssh mode, and my server requires an HTTP proxy to connect to the internet. I've added the HTTP proxy information in my settings.json file, and Copilot is working well, but copilot lab cannot connect. However, I've noticed that Copilot Lab doesn't seem to work with the proxy.

```
{
    "http.proxy": "xxx:3128",
    "http.proxySupport": "on",
    "http.proxyAuthorization": null,
    "http.proxyStrictSSL": false
}
```
I think Copilot Lab doesn't support proxies, is there any solution to use proxy for copilot lab? 

copilot lab give some err message 
```
[ERROR] [default] [2023-03-06T09:25:36.409Z] GitHub Copilot Labs could not connect to server. Extension activation failed: "connect ETIMEDOUT 20.205.243.168:443"
```

**Comment by jge162** [comment]
You can try adding a socks proxy to your settings.json file and/or try using a VPN.

`"network.proxy.type": 1,`
`"network.proxy.socks": "xxx",`
`"network.proxy.socks_port": xxx`
 
Please update xxx to your correct socks values. 

Please report back your findings. It is highly possible Copilot does not support this by the way.

**Reply by Novelfor** [reply]
this setting is not a legal setting in vscode... 
<img width="351" alt="image" src="https://user-images.githubusercontent.com/10961337/223154496-f11cae1f-953a-4cb9-bdda-d088e53a3bbb.png">

and I've tried using a socks proxy, but it doesn't work. the error message is same.

**Reply by liuxilai** [reply]
The same error in WSL, and copilot works fine.

`Something went wrong checking access: Failed to establish a socket connection to proxies: ["PROXY 127.0.0.1:7890"].`

**Comment by Novelfor** [comment]
For remote-ssh, set copilot-labs extensionKind to ui, then it will use local copilot-lab.
```
    "remote.extensionKind": {
        "GitHub.copilot-labs": ["ui"],
    }
```

**Reply by ProgramAlgo** [reply]
For use remote-ssh
In this way, it can be used on remote, and the problem of proxy can be solved temporarily

```
"remote.extensionKind": {
    "GitHub.copilot-labs": ["workspace"]
},

**Reply by kongmoumou** [reply]
nice, it works~

**Reply by fwarmuth** [reply]
For me this results in:
```
GitHub Copilot Labs could not connect to server. Extension activation failed: "unable to get local issuer certificate"
```
In my remote config i got, because i am behind a DPI Proxy:
```
    ...
    "http.proxyStrictSSL": false,
    ...
```
But when i use `"GitHub.copilot-labs": ["ui"],` its running my local machine and its seems that my vscode is not trusting the cert.
Any Idea?

**Comment by kanghaov** [comment]
I face the same bug, I use clash as my vpn to get github，even I tried to add the setting.json about 
`{
    "network.proxy.type": 1,
    "network.proxy.socks": "7890",
    "network.proxy.socks_port": 7890,
    "remote.extensionKind": {
        "GitHub.copilot-labs": ["workspace"]
    },
}`
It does not work
How should I do ,ple

**Reply by kanghaov** [reply]
I solved the prob·lem by  "remote.extensionKind": {
        "GitHub.copilot-labs": ["ui"],
    }

**Comment by p81sunshine** [comment]
I solve this problem by just add proxy setting in the vscode. Specificly, my http proxy address is xxx:7980, then I add this in the vscode setting file
